controllers/volumereplication_controller.go

```diff
@@ -127,6 +127,11 @@ func (r *VolumeReplicationReconciler) Reconcile(ctx context.Context, req ctrl.Re
 		if err != nil {
 			r.Log.Error(err, "task failed", "taskName", failedTask)
 		}
+	} else if instance.Spec.ImageState == replicationv1alpha1.Primary {
+		failedTask, err := r.markVolumeAsPrimary(volumeHandle, parameters, secret)
+		if err != nil {
+			r.Log.Error(err, "task failed", "taskName", failedTask)
```

Comment:
task failed but we are not return err to reconcile. in the above task also we need to fix it.